x-pack/plugins/apm/server/routes/service_map/get_service_map.ts

```diff
@@ -24,44 +17,41 @@ import {
 import { getServiceMapFromTraceIds } from './get_service_map_from_trace_ids';
 import { getTraceSampleIds } from './get_trace_sample_ids';
 import { transformServiceMapResponses } from './transform_service_map_responses';
-import { ENVIRONMENT_ALL } from '../../../common/environment_filter_values';
-import { getProcessorEventForTransactions } from '../../lib/helpers/transactions';
-import { ServiceGroup } from '../../../common/service_groups';
-import { serviceGroupQuery } from '../../lib/service_group_query';
 import { APMEventClient } from '../../lib/helpers/create_es_client/create_apm_event_client';
 import { APMConfig } from '../..';
+import { getServicesData } from './get_services_data';
 
 export interface IEnvOptions {
   mlClient?: MlClient;
   config: APMConfig;
   apmEventClient: APMEventClient;
-  serviceNames?: string[];
+  serviceName?: string;
   environment: string;
   searchAggregatedTransactions: boolean;
   logger: Logger;
   start: number;
   end: number;
-  serviceGroup: ServiceGroup | null;
+  serviceGroupKuery?: string;
 }
 
 async function getConnectionData({
   config,
   apmEventClient,
-  serviceNames,
+  serviceName,
   environment,
   start,
   end,
-  serviceGroup,
+  serviceGroupKuery,
 }: IEnvOptions) {
   return withApmSpan('get_service_map_connections', async () => {
     const { traceIds } = await getTraceSampleIds({
       config,
       apmEventClient,
-      serviceNames,
+      serviceName,
       environment,
       start,
       end,
-      serviceGroup,
+      serviceGroupKuery,
```

Comment:
We only use the `kuery` from the service group. Changing this to be an optional string is slightly simpler (and more clear).